### PR TITLE
Followup changes to 15817 (Segment schema publishing and polling)

### PR DESCRIPTION
**Pull request by findingrish**
Followup to https://github.com/apache/druid/pull/15817

Notable changes: 
* Rename DataSegmentWithSchema -> DataSegmentWithMetadata.
* Rename KillUnreferencedSegmentSchemaDuty -> KillUnreferencedSegmentSchema.
* Remove references to SMQ abbreviation. 
* Update naming for inTransit maps in `SegmentSchemaCache`. 
* Add delimiter in schema fingerprint hash computation.

### CODE REVIEW

File: docs/configuration/index.md [hunk 1]
```diff
@@ -899,6 +915,9 @@ These Coordinator static configurations can be defined in the `coordinator/runti
...
+|`druid.coordinator.kill.segmentSchema.on`| Boolean value for whether to enable automatic deletion of unused segment schemas. If set to true, Coordinator will periodically identify segment schemas which are not referenced by any used segment and mark them as unused. At a later point, these unused schemas are deleted. Only applies if [Centralized Datasource schema](#centralized-datasource-schema) feature is enabled. | No | True|
+|`druid.coordinator.kill.segmentSchema.period`| How often to do automatic deletion of segment schemas in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) duration format. Value must be equal to or greater than `druid.coordinator.period.metadataStoreManagementPeriod`. Only applies if `druid.coordinator.kill.segmentSchema.on` is set to true.| No| `P1D`|
+|`druid.coordinator.kill.segmentSchema.durationToRetain`| Duration of segment schemas to be retained from the time it was marked as unused in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) duration format. Only applies if `druid.coordinator.kill.segmentSchema.on` is set to true.| Yes, if `druid.coordinator.kill.segmentSchema.on` is set to true.| `P90D`|
```
Comment (317brian) [inline docs/configuration/index.md:920]:
```suggestion
|`druid.coordinator.kill.segmentSchema.on`| Boolean value for enabling automatic deletion of unused segment schemas. If set to true, the Coordinator service periodically identifies segment schemas that are not referenced by any used segment and marks them as unused. At a later point, these unused schemas are deleted. Only applies if the [Centralized Datasource schema](#centralized-datasource-schema) feature is enabled.| No | True|
|`druid.coordinator.kill.segmentSchema.period`| How often to do automatic deletion of segment schemas in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) duration format. Value must be equal to or greater than `druid.coordinator.period.metadataStoreManagementPeriod`. Only applies if `druid.coordinator.kill.segmentSchema.on` is set to true.| No| `P1D`|
|`druid.coordinator.kill.segmentSchema.durationToRetain`| [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) duration for the time a segment schema is retained for from when it's marked as unused. Only applies if `druid.coordinator.kill.segmentSchema.on` is set to true.| Yes, if `druid.coordinator.kill.segmentSchema.on` is set to true.| `P90D`|
```

**Review by cryptoe — APPROVED** [review]
Changes lgtm.

File: docs/configuration/index.md (hunk 1 above)
Comment (findingrish) [reply docs/configuration/index.md:920]: Hi @317brian, I missed pushing this change. I will raise a separate PR.

File: integration-tests/docker/druid.sh [hunk 2]
```diff
@@ -85,7 +85,7 @@ setupData()
...
+  if [ "$DRUID_INTEGRATION_TEST_GROUP" = "query" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "query-retry" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "query-error" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "high-availability" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "security" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "ldap-security" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "upgrade" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "centralized-datasource-schema" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "cds-task-schema-publish-disabled" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "cds-coordinator-metadata-query-disabled" ]; then
```
Comment (gianm) [inline integration-tests/docker/druid.sh:88]: This rename needs to be done in `standard-its.yml` as well.

**Pull request merged**